### PR TITLE
feat: [STO-6543]: anchore scanner

### PR DESCRIPTION
**Pull request by harness-dc**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53737)

### CODE REVIEW

File: 315-sto-manager/contracts/src/main/java/io/harness/sto/STOStepType.java [hunk 1]
```diff
@@ -89,6 +90,8 @@
 @AllArgsConstructor
 @OwnedBy(STO)
 public enum STOStepType {
+  ANCHORE("Anchore", "Anchore", FeatureName.STO_STEP_PALETTE_ANCHORE, AnchoreScanNode.class, EntityType.ANCHORE,
```
Comment (atmollo-harness) [inline 315-sto-manager/contracts/src/main/java/io/harness/sto/STOStepType.java:93]: This FF doesn't match the one that is already in FeatureName.java and UAT

File: 315-sto-manager/contracts/src/main/java/io/harness/beans/steps/nodes/security/AnchoreScanNode.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import static io.harness.annotations.dev.HarnessTeam.STO;
+
+import static com.fasterxml.jackson.annotation.JsonTypeInfo.As.EXTERNAL_PROPERTY;
+import static com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME;
+
+import io.harness.annotation.RecasterAlias;
+import io.harness.annotations.dev.OwnedBy;
+import io.harness.beans.steps.CIAbstractStepNode;
+import io.harness.beans.steps.CIStepInfoType;
+import io.harness.beans.steps.stepinfo.security.PrismaCloudStepInfo;
```
Comment (atmollo-harness) [inline 315-sto-manager/contracts/src/main/java/io/harness/beans/steps/nodes/security/AnchoreScanNode.java:19]: Don't want to be importing prisma cloud classes unless functionality is specifically being shared

File: 315-sto-manager/contracts/src/main/java/io/harness/beans/steps/stepinfo/security/shared/STOYamlAnchoreToolData.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import io.swagger.annotations.ApiModelProperty;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+@OwnedBy(STO)
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class STOYamlAnchoreToolData {
+    @ApiModelProperty(dataType = STRING_CLASSPATH, name = "product_domain_rbac") protected ParameterField<String> productDomainRbac;
```
Comment (atmollo-harness) [inline 315-sto-manager/contracts/src/main/java/io/harness/beans/steps/stepinfo/security/shared/STOYamlAnchoreToolData.java:27]: what is `product_domain_rbac` for?

File: 315-sto-manager/contracts/src/main/java/io/harness/beans/steps/nodes/security/AnchoreScanNode.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+public class AnchoreScanNode extends CIAbstractStepNode {
+    @JsonProperty("type") @NotNull AnchoreScanNode.StepType type = StepType.Anchore;
+    @NotNull
+    @JsonProperty("spec")
+    @JsonTypeInfo(use = NAME, property = "type", include = EXTERNAL_PROPERTY, visible = true)
+    AnchoreStepInfo stepInfo;
+
+    @Override
+    public String getType() {
+        return CIStepInfoType.ANCHORE.getDisplayName();
```
Comment (atmollo-harness) [inline 315-sto-manager/contracts/src/main/java/io/harness/beans/steps/nodes/security/AnchoreScanNode.java:48]: ANCHORE needs to be added to the CIStepInfoType enum

**Comment by harness-dc** [comment]
trigger uts

File: 332-ci-manager/service/src/test/resources/recaster/alias/alias.txt [hunk 5]
```diff
@@ -21,7 +21,7 @@ io.harness.ci.pipeline.executions.beans.CIScmDetails
 io.harness.beans.sweepingoutputs.CodeBaseConnectorRefSweepingOutput
 io.harness.yaml.core.failurestrategy.FailureStrategyConfig
 io.harness.beans.steps.nodes.ArtifactoryUploadNode
-io.harness.beans.steps.stepinfo.security.FossaStepInfo
+io.harness.beans.steps.stepinfo.security.FossaStepInfo1
```
Comment (atmollo-harness) [inline 332-ci-manager/service/src/test/resources/recaster/alias/alias.txt:24]: Is this a typo?

**Comment by harness-dc** [comment]
trigger uts

**Comment by harness-dc** [comment]
trigger ut0

**Comment by harness-dc** [comment]
trigger uts

**Review by atmollo-harness — APPROVED** [review]

**Review by r0610205 — APPROVED** [review]

**Comment by harness-dc** [comment]
trigger ut9

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**